tests/lib/rules/new-line-between-multi-line-property.js

```diff
@@ -0,0 +1,434 @@
+/**
+ * @fileoverview Enforce new lines between multi-line properties in Vue components.
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/new-line-between-multi-line-property')
+const RuleTester = require('eslint').RuleTester
+const ruleTester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: { ecmaVersion: 2015, sourceType: 'module' }
+})
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+ruleTester.run('new-line-between-multi-line-property', rule, {
+  valid: [
+    // test good example of proposal https://github.com/vuejs/eslint-plugin-vue/issues/391
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        name: 'component-name',
+        props: {
+          value: {
+            type: String,
+            required: true
+          },
+
+          focused: {
+            type: Boolean,
+            default: false
+          },
+
+          label: String,
+          icon: String
+        },
+
+        computed: {
+          formattedValue: function () {
+            // ...
+            // ...
+            // ...
+            // ...
+          },
+
+          inputClasses: function () {
+            // ...
+            // ...
+            // ...
+            // ...
+          }
+        },
+
+        methods: {
+          methodA: function () {
+            // ...
+            // ...
+            // ...
+          },
+
+          methodB: function () {
+            // ...
+            // ...
+            // ...
+          }
+        }
+      }
+      </script>
+      `
+    }
+  ],
+
+  invalid: [
```

Comment:
Please add test cases other than Vue components and test cases that use object expressions inside Vue components.